content/en/docs/tasks/administer-cluster/kubeadm/kubeadm-upgrade-ha-1-13.md

```diff
@@ -0,0 +1,233 @@
+---
+reviewers:
+- luxas
+- timothysc
+- jbeda
+title: Upgrading kubeadm HA clusters from v1.12 to v1.13
+content_template: templates/task
+---
+
+{{% capture overview %}}
+
+This page explains how to upgrade a highly available (HA) Kubernetes cluster created with `kubeadm` from version 1.12.x to version 1.13.y. In addition to upgrading, you must also follow the instructions in [Creating HA clusters with kubeadm](/docs/setup/independent/high-availability/).
+
+{{% /capture %}}
+
+{{% capture prerequisites %}}
+
+Before proceeding:
+
+- You need to have a `kubeadm` HA cluster running version 1.12 or higher.
+- Make sure you read the [release notes](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG-1.13.md) carefully.
+- Make sure to back up any important components, such as app-level state stored in a database. `kubeadm upgrade` does not touch your workloads, only components internal to Kubernetes, but backups are always a best practice.
+- Check the prerequisites for [Upgrading/downgrading kubeadm clusters between v1.12 to v1.13](/docs/tasks/administer-cluster/kubeadm/kubeadm-upgrade-1-13/).
+
+{{< note >}}
+All commands on any control plane or etcd node should be run as root.
+{{< /note >}}
+
+{{% /capture %}}
+
+{{% capture steps %}}
+
+## Prepare for both methods
+
+Upgrade `kubeadm` to the version that matches the version of Kubernetes that you are upgrading to:
+
+```shell
+apt-mark unhold kubeadm && \
+apt-get update && apt-get install -y kubeadm && \
+apt-mark hold kubeadm
+```
+
+Check prerequisites and determine the upgrade versions:
+
+```shell
+kubeadm upgrade plan
+```
+
+You should see something like the following:
+
+```
+Upgrade to the latest version in the v1.13 series:
+
+COMPONENT            CURRENT   AVAILABLE
+API Server           v1.12.2   v1.13.0
+Controller Manager   v1.12.2   v1.13.0
+Scheduler            v1.12.2   v1.13.0
+Kube Proxy           v1.12.2   v1.13.0
+CoreDNS              1.2.2     1.2.6
+```
+
+## Stacked control plane nodes
+
+### Upgrade the first control plane node
+
+Modify `configmap/kubeadm-config` for this control plane node:
```

Comment:
@rdodev I'm not sure I fully understand why those changes are necessary and what is the expected results.

From my tests etcd manifest is not touched by upgrade because the etcd version does not change. 

But this is not a problem unless you are planning to join additional control-plane nodes; this is possible only if a control plane address is set, and it requires additional steps documented here https://github.com/kubernetes/kubeadm/issues/1269#issuecomment-441116249